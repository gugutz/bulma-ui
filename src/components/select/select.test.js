import React from 'react'
import renderer from 'react-test-renderer'
import Select from './select'

describe('<Select /> component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Select>
          <option>Select dropdown</option>
          <option>With options</option>
        </Select>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should support isRounded prop', () => {
    const tree = renderer
      .create(
        <Select isRounded>
          <option>Select dropdown</option>
          <option>With options</option>
        </Select>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should support multiple prop with a value', () => {
    const tree = renderer
      .create(
        <Select multiple="8">
          <option value="Argentina">Argentina</option>
          <option value="Bolivia">Bolivia</option>
          <option value="Brazil">Brazil</option>
          <option value="Suriname">Suriname</option>
          <option value="Uruguay">Uruguay</option>
          <option value="Venezuela">Venezuela</option>
        </Select>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should support multiple prop as boolean', () => {
    const tree = renderer
      .create(
        <Select multiple>
          <option value="Argentina">Argentina</option>
          <option value="Bolivia">Bolivia</option>
          <option value="Brazil">Brazil</option>
          <option value="Suriname">Suriname</option>
          <option value="Uruguay">Uruguay</option>
          <option value="Venezuela">Venezuela</option>
        </Select>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should support other props from select', () => {
    const tree = renderer
      .create(
        <Select name="email">
          <option>Select dropdown</option>
          <option>With options</option>
        </Select>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  const colors = ['primary', 'info', 'success', 'warning', 'danger', 'default']

  for (const color of colors) {
    it('should support color prop with ' + color, () => {
      const tree = renderer
        .create(
          <Select color={color}>
            <option>Select dropdown</option>
            <option>With options</option>
          </Select>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  }

  const states = ['hovered', 'focused', 'normal', 'loading']

  for (const state of states) {
    it('should support state prop with ' + state, () => {
      const tree = renderer
        .create(
          <Select state={state}>
            <option>Select dropdown</option>
            <option>With options</option>
          </Select>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  }

  const sizes = ['small', 'normal', 'medium', 'large']

  for (const size of sizes) {
    it('should support size prop with ' + size, () => {
      const tree = renderer
        .create(
          <Select size={size}>
            <option>Select dropdown</option>
            <option>With options</option>
          </Select>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  }
})
