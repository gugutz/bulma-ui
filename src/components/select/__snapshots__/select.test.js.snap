// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> component should render correctly 1`] = `
<div
  className="select"
>
  <select
    className=""
    multiple={false}
    size={false}
  >
    <option>
      Select dropdown
    </option>
    <option>
      With options
    </option>
  </select>
</div>
`;

exports[`<Select /> component should support color prop with danger 1`] = `
<div
  className="select is-danger"
>
  <select
    className=""
    multiple={false}
    size={false}
  >
    <option>
      Select dropdown
    </option>
    <option>
      With options
    </option>
  </select>
</div>
`;

exports[`<Select /> component should support color prop with default 1`] = `
<div
  className="select"
>
  <select
    className=""
    multiple={false}
    size={false}
  >
    <option>
      Select dropdown
    </option>
    <option>
      With options
    </option>
  </select>
</div>
`;

exports[`<Select /> component should support color prop with info 1`] = `
<div
  className="select is-info"
>
  <select
    className=""
    multiple={false}
    size={false}
  >
    <option>
      Select dropdown
    </option>
    <option>
      With options
    </option>
  </select>
</div>
`;

exports[`<Select /> component should support color prop with primary 1`] = `
<div
  className="select is-primary"
>
  <select
    className=""
    multiple={false}
    size={false}
  >
    <option>
      Select dropdown
    </option>
    <option>
      With options
    </option>
  </select>
</div>
`;

exports[`<Select /> component should support color prop with success 1`] = `
<div
  className="select is-success"
>
  <select
    className=""
    multiple={false}
    size={false}
  >
    <option>
      Select dropdown
    </option>
    <option>
      With options
    </option>
  </select>
</div>
`;

exports[`<Select /> component should support color prop with warning 1`] = `
<div
  className="select is-warning"
>
  <select
    className=""
    multiple={false}
    size={false}
  >
    <option>
      Select dropdown
    </option>
    <option>
      With options
    </option>
  </select>
</div>
`;

exports[`<Select /> component should support isRounded prop 1`] = `
<div
  className="select is-rounded"
>
  <select
    className=""
    multiple={false}
    size={false}
  >
    <option>
      Select dropdown
    </option>
    <option>
      With options
    </option>
  </select>
</div>
`;

exports[`<Select /> component should support multiple prop as boolean 1`] = `
<div
  className="select is-multiple"
>
  <select
    className=""
    multiple={true}
    size={true}
  >
    <option
      value="Argentina"
    >
      Argentina
    </option>
    <option
      value="Bolivia"
    >
      Bolivia
    </option>
    <option
      value="Brazil"
    >
      Brazil
    </option>
    <option
      value="Suriname"
    >
      Suriname
    </option>
    <option
      value="Uruguay"
    >
      Uruguay
    </option>
    <option
      value="Venezuela"
    >
      Venezuela
    </option>
  </select>
</div>
`;

exports[`<Select /> component should support multiple prop with a value 1`] = `
<div
  className="select is-multiple"
>
  <select
    className=""
    multiple={true}
    size="8"
  >
    <option
      value="Argentina"
    >
      Argentina
    </option>
    <option
      value="Bolivia"
    >
      Bolivia
    </option>
    <option
      value="Brazil"
    >
      Brazil
    </option>
    <option
      value="Suriname"
    >
      Suriname
    </option>
    <option
      value="Uruguay"
    >
      Uruguay
    </option>
    <option
      value="Venezuela"
    >
      Venezuela
    </option>
  </select>
</div>
`;

exports[`<Select /> component should support other props from select 1`] = `
<div
  className="select"
>
  <select
    className=""
    multiple={false}
    name="email"
    size={false}
  >
    <option>
      Select dropdown
    </option>
    <option>
      With options
    </option>
  </select>
</div>
`;

exports[`<Select /> component should support size prop with large 1`] = `
<div
  className="select is-large"
>
  <select
    className=""
    multiple={false}
    size={false}
  >
    <option>
      Select dropdown
    </option>
    <option>
      With options
    </option>
  </select>
</div>
`;

exports[`<Select /> component should support size prop with medium 1`] = `
<div
  className="select is-medium"
>
  <select
    className=""
    multiple={false}
    size={false}
  >
    <option>
      Select dropdown
    </option>
    <option>
      With options
    </option>
  </select>
</div>
`;

exports[`<Select /> component should support size prop with normal 1`] = `
<div
  className="select"
>
  <select
    className=""
    multiple={false}
    size={false}
  >
    <option>
      Select dropdown
    </option>
    <option>
      With options
    </option>
  </select>
</div>
`;

exports[`<Select /> component should support size prop with small 1`] = `
<div
  className="select is-small"
>
  <select
    className=""
    multiple={false}
    size={false}
  >
    <option>
      Select dropdown
    </option>
    <option>
      With options
    </option>
  </select>
</div>
`;

exports[`<Select /> component should support state prop with focused 1`] = `
<div
  className="select"
>
  <select
    className="is-focused"
    multiple={false}
    size={false}
  >
    <option>
      Select dropdown
    </option>
    <option>
      With options
    </option>
  </select>
</div>
`;

exports[`<Select /> component should support state prop with hovered 1`] = `
<div
  className="select"
>
  <select
    className="is-hovered"
    multiple={false}
    size={false}
  >
    <option>
      Select dropdown
    </option>
    <option>
      With options
    </option>
  </select>
</div>
`;

exports[`<Select /> component should support state prop with loading 1`] = `
<div
  className="select is-loading"
>
  <select
    className=""
    multiple={false}
    size={false}
  >
    <option>
      Select dropdown
    </option>
    <option>
      With options
    </option>
  </select>
</div>
`;

exports[`<Select /> component should support state prop with normal 1`] = `
<div
  className="select"
>
  <select
    className=""
    multiple={false}
    size={false}
  >
    <option>
      Select dropdown
    </option>
    <option>
      With options
    </option>
  </select>
</div>
`;
